fix: unexpected decoding behavior due to cheerio package version (#6)

Mister-Kin opened the pull request:
Update cheerio version to latest v1.0.0 can resolve this problem.

Also remove value 'excerpt' from variable 'contentKeys'. It might lead to render blank content for post excerpt in Home Page when using hexo-theme-next v8.22.0.

mssonicbld commented:
/azp run

r12f commented:
Hi @Mister-Kin , thanks a lot for the PR! Do you mind to split the change into 2? One for cheerio and one for the removal. I might need to check the latter one, but at least we can get the first part in soon.

Mister-Kin commented:
> Hi @Mister-Kin , thanks a lot for the PR! Do you mind to split the change into 2? One for cheerio and one for the removal. I might need to check the latter one, but at least we can get the first part in soon.

My bad. It really should have been separated.

For the current modification I have only tested hexo-theme-next v8.22.0 to be compatible to run.

I'll test a few other themes for compatibility later. If there are no problems, I will resubmit this PR.

Mister-Kin committed:
fix: unexpected decoding behavior due to cheerio package version (#6)

Update cheerio version to latest v1.0.0 can resolve this problem.

Mister-Kin committed:
fix: sometimes render blank content for post excerpt and cheerio always still decodes html entities

Firstly, by default, parse5 using by cheerio treats documents it receives as full HTML documents. This mode will add nested <head> and <body> tags to the data which might lead to render blank content for post excerpt because of the wrong nested tags.

Secondly, using `decodeEntities: false` in load function of cheerio seems not working in cheerio v1.0.0. Cheerio still decodes html entities even using `decodeEntities: false`.

Setting to use htmlparser2 for cheerio can resolve this two problems which won't add any redundant tags to the data and won't decode html entities for the data.

Setting to fragment Mode for cheerio can also resolve the first problem but unnecessarily. Because using htmlparser2 for cheerio already resolves two of them.

mssonicbld commented:
/azp run

Mister-Kin commented:
Original configuration for cheerio doesn't work well with the new cheerio version.

## Original configuration for cheerio load method
Original configuration for cheerio load method of `updateHeadingIndexes` function in `/lib/index.js`
```javascript
var $ = cheerio.load(data, { decodeEntities: false });
```
It leads to two problems.

Add nested `<head>` and `<body>` tags to the data which might lead to render blank content for post excerpt because of the wrong nested tags.
![初始设置](https://github.com/user-attachments/assets/4a802a7b-3b12-4d79-9484-8f84377b1679)

Cheerio still decodes html entities even using `decodeEntities: false`.
![decodefail](https://github.com/user-attachments/assets/fb987c87-7165-4d2f-909f-5011e1034e2a)


## New configuration for cheerio load method
Just see commit [50ed3f4](https://github.com/r12f/hexo-heading-index/pull/8/commits/50ed3f449bb8d43e998ca31a6af01ac7c33ace22).
```javascript
var $ = cheerio.load(data, { xml: { xmlMode: false, decodeEntities: false } })
```
No more redundant tags for the data.
![修正参数作为片段解析](https://github.com/user-attachments/assets/c520b925-00b2-45fa-aa70-83f1badbb94e)

Won't decode html entities for the data.
![decodesuccess](https://github.com/user-attachments/assets/2825239f-ff04-4682-98e0-2e1f726efc4f)

## Testing the code of new commit
The effect of default theme `landscape` of hexo-cli.
![good_to_go](https://github.com/user-attachments/assets/71de56f2-118e-48e3-aa7a-12bcdaa69bc3)

The effect of my next theme.
![good_to_go2](https://github.com/user-attachments/assets/efcbf180-1f5e-4555-8a60-202955bb9614)

I think the code is good to go. @r12f
